Pull out path constants from imager into yml file

aheschl1 opened the pull request:
Paths to binaries were hardcoded in imager. They have been pulled out into a yml file. At build time, the file is parsed and imported as constants into the imager code.

Furthermore, the folder name for the output of the images is included in the config file, as well as the path to static files to be included.

Issue: https://github.com/moturus/motor-os/issues/24

aheschl1 committed:
Pull out path constants from imager into yml file

Paths to binaries were hardcoded in imager.
They have been pulled out into a yml file.
At build time, the file is parsed and imported
as constants into the imager code.

aheschl1 commented:
Approach taken is to expose the same constants as currently exist, though the build.rs file, which parses the yaml.

Introduces new dependency: rust-yaml

lasiotus commented:
Thanks a lot for the PR! I was actually imagining processing a config file in the main.rs, not in build.rs. Why do you think processing the config file in build.rs is better vs at runtime?

The benefits of having a "runtime" config (processed in main.rs) is that a single imager binary can build different images, based on different config files. So in our current setup, imager will be invoked (via Makefile) three times, once with full.yml, once with web.yml, and once with motor-fs.yml configs (so the config file describes input binaries and the file system to use). In the future I can imagine the same imager binary used to package a "test" image that will run tests "on boot".

aheschl1 commented:
Thanks for the comment - makes sense to me to do it at runtime since you are going for one build capable of generating multiple images. I had originally done it at build time as I thought of it more as easily configurable constants that won't change.

I've pulled it into runtime, and added more complex "syntax" for the config file.

Some names should probably be changed so let me know what changes you'd have in mind. Maybe config structure as well if you don't like the format. For one, would you prefer `part3_fs` just to be 'fs_format' because partition 3 is the only one with a filesystem?

I have a couple questions. (I am very much a novice when it comes to OS topics)
1. How do `srfs`, `flatfs`, and `motor-fs` filesystems work, how do they differ?
2. What is on the _part1_ and _part2_ partitions?
3. Importantly: **what steps would you recommend taking to assert the correctness of this PR?** do you have any routines for validating build results?

aheschl1 committed:
Add argument to specify image file in run-qemu.sh

In order to use ./run-qemu.sh to launch a specific
image file, one needed to modify the script to use
a different hardcoded file name. This commit adds
an optional --img=<image_file> argument to the script.
